scripts/asr_language_modeling/ngram_lm/kenlm_utils.py

```diff
@@ -17,98 +17,244 @@
 
             tokenized_data = parallel(
-                delayed(tokenize_str)(data[start : start + chunk_size], tokenizer, token_offset)
+                delayed(tokenize_str)(data[start : start + chunk_size], tokenizer)
                 for start in range(start, end, chunk_size)
             )
 
             # Write dataset
             write_dataset(tokenized_data, path)
             current_step += len(tokenized_data)
-            print(f"Finished writing {len(tokenized_data)} chunks to {path}. Current chunk index = {current_step}")
+            if path:
+                print(f"Finished writing {len(tokenized_data)} chunks to {path}. Current chunk index = {current_step}")
             del tokenized_data
             if end >= dataset_len:
                 break
 
 
 def write_dataset(chunks, path):
-    basedir = os.path.dirname(path)
-
-    if not os.path.exists(basedir):
-        os.makedirs(basedir, exist_ok=True)
-
-    with open(path, 'a+', encoding='utf-8') as f:
-        for chunk_idx in tqdm(range(len(chunks)), desc='Chunk ', total=len(chunks), unit=' chunks'):
+    if path:
+        basedir = os.path.dirname(path)
+        if not os.path.exists(basedir):
+            os.makedirs(basedir, exist_ok=True)
+        with open(path, 'a+', encoding='utf-8') as f:
+            for chunk_idx in tqdm(range(len(chunks)), desc='Chunk ', total=len(chunks), unit=' chunks'):
+                for text in chunks[chunk_idx]:
+                    line = ' '.join(text)
+                    f.write(f"{line}\n")
+    else:
+        for chunk_idx in range(len(chunks)):
             for text in chunks[chunk_idx]:
                 line = ' '.join(text)
-                f.write(f"{line}\n")
+                print(f"{line}\n")
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description="Avg pytorch weights")
+    parser.add_argument(
+        "--train_path",
+        required=True,
+        nargs="+",
+        type=str,
+        help="Path to the training file or whitespace separated files. it can be a text file, JSON manifest or .json.gz",
+    )
+    parser.add_argument(
+        "--tokenizer_model_file",
+        required=True,
+        type=str,
+        help="The path to '.model' file of the SentencePiece tokenizer, or '.nemo' file of the ASR model, or name of a pretrained NeMo model",
+    )
+    parser.add_argument(
+        "--do_lowercase", action='store_true', help="Whether to apply lower case conversion on the training text"
+    )
+    parser.add_argument(
+        "--rm_punctuation", action='store_true', help="Whether to remove punctuation marks from text",
+    )
+    parser.add_argument(
+        "--separate_punctuation",
+        action='store_true',
+        help="Whether to separate punctuation with the previouse word by space when --clean_text and --punctuation_to_preserveis is used",
+    )
+
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
```

Comment:
Is user going to run this file independently?

Reply:
This file was a utils file actually. Couldn't we just call the method iter_files from train_kenlm.py and avoid creating subprocesses?

Reply:
done